### PR TITLE
Ai combat replay decorations

### PR DESCRIPTION
**Pull request by Zerthox**
**Common**
- [x] Arrow indicators (as rectangle) for all dashes & line attacks in Air, Fire, Dark
- [x] Spread circles around player
  * [x] Indicator
  * [x] Air Detonate
  * [x] Fire Detonate
  * [x] Water Detonate
  * [x] Dark Detonate
- [x] Orbs with red border
  * [x] Air
  * [x] Fire
  * [x] Water
  * [x] Dark (purple)
- [x] Cone frontal & regular circular AoE around Ai
- [x] Detonating ground circle patterns
  * [x] Air
  * [x] Fire
  * [x] Water
  * [x] Dark
- [x] Agony applying red circular spin
- [x] Pulsing ground circle patterns
  * [x] Air
  * [x] Fire
  * [x] Water
  * [x] Dark

**Air**
- [x] Lightning Strike indicators

**Fire**
- [x] Meteor indicators
- [ ] Fallen Meteor display (safety zone?)
- [ ] Fallen Meteor despawn explosion

**Water**
- [x] Tethers between Enraged Water Sprite & player
- [x] Green circles attached to player
- [x] Tornado (Water line attack) indicators

**Dark**
- [x] Sorrow attack indicators
- [x] Guilt tether to player

### CODE REVIEW

**Comment by EliphasNUIT** [comment]
You think this is mergeable? If you don't have the time to finish, you can add them in a TODO list on https://github.com/baaron4/GW2-Elite-Insights-Parser/issues/615 and we can merge what is currently done.

**Comment by Zerthox** [comment]
I'll try to complete a little more and then we can merge and move the rest to TODO.

**Comment by Linkaaaaa** [comment]
I can take over once merged and see what else can be added

**Comment by Zerthox** [comment]
Marking this as ready for review, would be maybe be a good idea to test it on some more logs before release.

File: GW2EIEvtcParser/ParserHelpers/SkillIDs.cs [hunk 1]
```diff
@@ -3319,6 +3320,7 @@ public static class SkillIDs
         public const long ElementalManipulationFire2 = 61256;
         public const long EmpathicManipulation7 = 61260;
         public const long ElementalManipulationFire3 = 61271;
+        public const long OverwhelmingSorrowHit = 61241;
```
Comment (EliphasNUIT) [inline GW2EIEvtcParser/ParserHelpers/SkillIDs.cs:3323]: Not correctly ordered.

File: GW2EIEvtcParser/ParserHelpers/SkillIDs.cs [hunk 2]
```diff
@@ -3380,6 +3382,8 @@ public static class SkillIDs
         public const long EmpathicManipulationGuilt = 61602;
         public const long EmpathicManipulation2 = 61604;
         public const long EmpathicManipulationSorrow = 61606;
+        public const long FixatedEnragedWaterSprite = 61615;
+        public const long FixatedGuilt = 61225;
```
Comment (EliphasNUIT) [inline GW2EIEvtcParser/ParserHelpers/SkillIDs.cs:3386]: Not correctly ordered.

File: GW2EIEvtcParser/EncounterLogic/Fractals/SunquaPeak/AiKeeperOfThePeak.cs [hunk 3]
```diff
@@ -400,12 +400,350 @@ protected override void SetInstanceBuffs(ParsedEvtcLog log)
...
+        {
+            base.ComputeNPCCombatReplayActors(target, log, replay);
+
+
+            IReadOnlyList<AnimatedCastEvent> casts = log.CombatData.GetAnimatedCastData(target.AgentItem);
+            switch (target.ID)
+            {
+                case (int)TrashID.CCSorrowDemon:
+                    {
+                        const long fullCastDuration = 11840;
```
Comment (EliphasNUIT) [inline GW2EIEvtcParser/EncounterLogic/Fractals/SunquaPeak/AiKeeperOfThePeak.cs:441]: The const here should be renamed to something that indicates that they are linked to Overwhelming sorrow

File: GW2EIEvtcParser/EncounterLogic/Fractals/SunquaPeak/AiKeeperOfThePeak.cs [hunk 4]
```diff
@@ -400,12 +400,350 @@ protected override void SetInstanceBuffs(ParsedEvtcLog log)
...
+                        const long hitDelay = 400;
+                        const uint indicatorSize = 2000;
+                        var detonates = casts.Where(x => x.SkillId == OverwhelmingSorrowDetonate).ToList();
+                        foreach (AnimatedCastEvent cast in casts)
+                        {
+                            switch (cast.SkillId)
```
Comment (EliphasNUIT) [inline GW2EIEvtcParser/EncounterLogic/Fractals/SunquaPeak/AiKeeperOfThePeak.cs:447]: why a switch?

File: GW2EIEvtcParser/EncounterLogic/Fractals/SunquaPeak/AiKeeperOfThePeak.cs [hunk 5]
```diff
@@ -400,12 +400,350 @@ protected override void SetInstanceBuffs(ParsedEvtcLog log)
...
+        {
+            // arrow indicators
+            if (log.CombatData.TryGetEffectEventsByGUID(EffectGUIDs.AiArrowAttackIndicator, out IReadOnlyList<EffectEvent> arrows))
+            {
+                const uint width = 80;
+                const uint length = 360;
+                foreach (EffectEvent effect in arrows)
+                {
+                    long duration = effect.Duration != 0 ? effect.Duration : 1800;
+                    (long, long) lifespan = (effect.Time, effect.Time + duration);
```
Comment (EliphasNUIT) [inline GW2EIEvtcParser/EncounterLogic/Fractals/SunquaPeak/AiKeeperOfThePeak.cs:500]: effect.ComputeLifespan(log, 1800);

**Comment by EliphasNUIT** [comment]
Thanks!

**Pull request merged**